### PR TITLE
GPUManger destructor does not throw exceptions with cudaSetDevice

### PR DESCRIPTION
**Pull request by naoyam**

### CODE REVIEW

**Review by benson31 — APPROVED** [review]
Looks good to me. But also, why did travis start running??

**Comment by naoyam** [comment]
Thanks. I don't know about Travis.

**Comment by naoyam** [comment]
I don't have write access, so please merge this if it's not a problem.

**Pull request merged**
